app/routes/session.py

```diff
@@ -169,9 +181,26 @@ def _set_questionnaire_supplementary_data(
         survey_id=metadata["survey_id"],
         sds_dataset_id=new_sds_dataset_id,
     )
+    _validate_supplementary_data_lists(
+        supplementary_data=supplementary_data["data"], schema=schema
+    )
     questionnaire_store.set_supplementary_data(supplementary_data["data"])
 
 
+def _validate_supplementary_data_lists(
+    *, supplementary_data: dict, schema: QuestionnaireSchema
+) -> None:
+    """
+    Validates that any lists the schema requires (which are those in the supplementary_data.lists property)
+    are included in the supplementary data
+    """
+    supplementary_lists = set(supplementary_data.get("items", {}).keys())
```

Comment:
Minor, either drop `set` as `keys` returns a view object which support set operations. Alternatively, drop `keys()` and keep `set` as in Py3 you don't need to say `keys()` in this case.

Reply:
Oh nice - didn't realise keys would support it out the box, done 👍